### PR TITLE
Remove unused functions and increase test coverage

### PR DESCRIPTION
**Pull request by cjprybol**

### CODE REVIEW

File: test/datatable.jl [hunk 1]
```diff
@@ -341,4 +341,46 @@ module TestDataTable
...
+        dt = DataTable(A = 1:10)
+        @test head(dt) == DataTable(A = 1:6)
+        @test head(dt, 1) == DataTable(A = 1)
+        @test tail(dt) == DataTable(A = 5:10)
+        @test tail(dt, 1) == DataTable(A = 10)
+    end
+
+    @testset "misc" begin
+        dt = DataTable(A = 'A':'C')
+        @test sprint(dump, dt) == "DataTables.DataTable  3 observations of 1 variables\n  A: Array{Char}((3,))\n    1: Char A\n    2: Char B\n    3: Char C\n"
```
Comment (nalimilan) [inline test/datatable.jl]: You can use triple quotes for this kind of multiline string:
```julia
"""DataTables.DataTable  3 observations of 1 variables
  A: Array{Char}((3,))
    1: Char A
    2: Char B
    3: Char C
"""
```

Comment (ararslan) [reply test/datatable.jl]: The leading newline is chomped out I believe, so I think you can add a line break between `"""` and `D`.

Comment (nalimilan) [reply test/datatable.jl]: No, it took me some time to get it right, but this one should work. OTOH leading spaces are removed when they align with the quotes.

Comment (ararslan) [reply test/datatable.jl]: Huh, TIL.

**Comment by cjprybol** [comment]
This should be ready to go

**Comment by cjprybol** [comment]
I've redone the commits here to be 4 logical units.

**Review by ararslan — APPROVED** [review]

**Comment by cjprybol** [comment]
Thanks!

**Pull request merged**